src/wallet/rpcwallet.cpp

```diff
@@ -2575,6 +2577,261 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
+                            "User can specify totalFee, or use RPC setpaytxfee to set a higher fee rate.\n"
+                            "\nArguments:\n"
+                            "1. \"txid\"              (string, required) The txid to be bumped\n"
+                            "2. output                (int, required) The output to be decremented\n"
+                            "3. options               (object, optional)\n"
+                            "   {\n"
+                            "     \"confTarget\"      (int, optional) Confirmation target (in blocks)\n"
+                            "     \"totalFee\"        (numeric, optional) Total fee to pay, in satoshis (not btc)\n"
+                            "     \"maxFee\"          (numeric, optional) Allows user to override bumpfee's max fee\n"
+                            "   }\n"
+                            "\nResult:\n"
+                            "{\n"
+                            "  \"txid\":    \"value\", (string)  The id of the new transaction\n"
+                            "  \"oldfee\":    n,         (numeric) Fee of the replaced transaction\n"
+                            "  \"fee\":       n,         (numeric) Fee of the new transaction\n"
+                            "}\n"
+                            "\nExamples:\n"
+                            "\nBump the fee, get the new transaction\'s txid\n"
+                            + HelpExampleCli("bumpfee", "<txid> <output>")
+                            );
+
+    RPCTypeCheck(request.params, boost::assign::list_of(UniValue::VSTR)(UniValue::VNUM));
+    uint256 hash;
+    hash.SetHex(request.params[0].get_str());
+    std::string strError;
+
+    // retrieve the original tx from the wallet
+    assert(pwalletMain != NULL);
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+    if (!pwalletMain->mapWallet.count(hash))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid or non-wallet transaction id");
+    const CWalletTx& wtx = pwalletMain->mapWallet[hash];
+
+    // check that we are not trying to bump a tx that has already been mined
+    if (wtx.GetDepthInMainChain() > 0)
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "This transaction has already been mined!");
+
+    // check that original tx signals opt-in-RBF
+    if (!SignalsOptInRBF(wtx))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Transaction is not bip-125 replaceable");
+
+    // Check that if the original tx has wallet conflicts (meaning there is at least one other tx
+    // in the wallet that spends at least one of the same outputs) that it is the most recent one.
+    // This prevents us from trying to bump fee on a tx that has already been bumped.  So if the
+    // user created txid1, bumped the fee which resulted in txid2, and now wishes to bump the fee
+    // again, the user must call bumpfee on txid2.
+    const CWalletTx *conflictTx;
```

Comment:
I don't think this section makes sense.  Transactions which are in newer in the wallet are not a good indication of the most "up to date" spend of the outputs.  If anything, conflicts in the mempool would be a good proxy, because at least for those you'd be able to calculate any descendants fees.  But it may make the most sense to make this more of a utility function and just assume the user is trying to bump the right txid.  I would just eliminate this set of checks entirely.